db-service/test/cqn4sql/search.test.js

```diff
@@ -120,4 +120,17 @@ describe('Replace attribute search by search predicate', () => {
       search((books.createdBy, books.modifiedBy, books.anotherText, books.title, books.descr, books.currency_code, books.dedication_text, books.dedication_sub_foo, books.dedication_dedication), ('x' OR 'y')) `,
     )
   })
+  it('Search with aggregated column and groupby must be put into having', () => {
+    // if we search on aggregated results, the search must be put into the having clause
+    const { Books } = cds.entities
+    let query = SELECT.from(Books)
+      .columns({ args: [{ ref: ['title'] }], as: 'firstInAlphabet', func: 'MIN' })
```

Comment:
Can you add some other aggregation to ensure that they are not searched?

Reply:
all `.funcs` are put into the search if the query result is grouped, `search.js`:
```js
if (cqn._aggregated || /* new parser */ cqn.SELECT.groupBy || cqn.SELECT?.from?.SELECT?.groupBy) {
    cqn.SELECT.columns &&
      cqn.SELECT.columns.forEach(column => {
        if (column.func) {
          // exclude $count by SELECT of number of Items in a Collection
          if (
            cqn.SELECT.columns.length === 1 &&
            column.func === 'count' &&
            (column.as === '_counted_' || column.as === '$count')
          ) {
            return
          }

          toBeSearched.push(column)
          return
        }
…
```

Only the ANSI SQL aggregations are put into the `having` clause though. Do you want me to add a test for aggregations which are _not_ put into the `having` clause?

Reply:
No, I meant to add an aggregation in the test that is not string like which should be excluded, e. g. sum(price) as price

Reply:
okay sure, then I need to add a list of aggregations which must be included from being searched first.